### PR TITLE
initial testing setup

### PR DESCRIPTION
**Pull request by cryptix**
How do you guys feel about this testing setup?

Using the example data feels a bit wonky. For Travis we should probably use a specific version of the docker image then.

@jbenet i couldn't enable travis for this repo because i don't have admin rights.

### CODE REVIEW

**Comment by cryptix** [comment]
@jbenet @whyrusleeping rebased and fixed :) :+1:

**Comment by whyrusleeping** [comment]
awesome! let me enable travis real quick...

**Comment by jbenet** [comment]
travis enabled

**Comment by jbenet** [comment]
Will probably need to repush to trigger it

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  allow_failures:
+    - go: tip
+
+services:
+  - docker
+
+before_install:
+- docker pull jbenet/go-ipfs:latest
+- mkdir /tmp/ipfs && chmod 0777 /tmp/ipfs
+- docker run -d -v /tmp/ipfs:/data/ipfs -p 8080:8080 -p 4001:4001 -p 5001:5001 jbenet/go-ipfs
```
Comment (jbenet) [inline .travis.yml]: oh wow, this works? nice!!

Comment (cryptix) [inline .travis.yml]: Yup! I use it in my git-remote-ipfs tests :)

**Comment by cryptix** [comment]
@jbenet ping :)

**Comment by jbenet** [comment]
LGTM! :+1:

**Comment by jbenet** [comment]
somehow the merge testing failed: https://travis-ci.org/ipfs/go-ipfs-api/builds/81770162

**Comment by cryptix** [comment]
> somehow the merge testing failed: https://travis-ci.org/ipfs/go-ipfs-api/builds/81770162

Hrm.. they did pass before... `Error response from daemon: Cannot start container ....: iptables failed: iptables --wait -t filter -A DOCKER ! -i docker0 -o docker0 -p tcp -d 172.17.0.1 --dport 8080 -j ACCEPT: iptables: No chain/target/match by that name.` sadly isn't code related at all. I've seen it a couple of times but usually re-run fixed it after the 2nd try, now I can't get it to pass at all.. I fear it's influenced by load... :-/

**Comment by cryptix** [comment]
Related: https://github.com/travis-ci/travis-ci/issues/4778

**Pull request merged**